osu.Game/Online/API/Requests/Responses/APIScoreInfo.cs

```diff
@@ -34,17 +93,19 @@ public ScoreInfo CreateScoreInfo(RulesetStore rulesets)
                 MaxCombo = MaxCombo,
                 User = User,
                 Accuracy = Accuracy,
-                OnlineScoreID = OnlineScoreID,
+                OnlineScoreID = OnlineID,
                 Date = Date,
                 PP = PP,
-                BeatmapInfo = BeatmapInfo,
                 RulesetID = OnlineRulesetID,
                 Hash = Replay ? "online" : string.Empty, // todo: temporary?
                 Rank = Rank,
                 Ruleset = ruleset,
                 Mods = mods,
             };
 
+            if (beatmap != null)
```

Comment:
Think it may be worth having a consistency check here? As in if `Beatmap` (the deserialised one) isn't null too and differs from the passed `beatmap` (probably checked by comparing online ID) then throw an exception or something?

Reply:
Not sure this will bring much value.. the end goal is to (hopefully) remove this method entirely. The use case for this parameter is also only used for display purposes, so it can't negatively affect online systems or score storage.